gh-102780: Fix uncancel() call in asyncio timeouts (#102815)
python/cpython: Doc/library/asyncio-task.rst

It is recommended that coroutines use ``try/finally`` blocks to robustly
perform clean-up logic. In case :exc:`asyncio.CancelledError`
is explicitly caught, it should generally be propagated when

> clean-up is complete. :exc:`asyncio.CancelledError` is a subclass of

Reviewer: Strictly speaking _all_ exceptions are "a subclass" of `BaseException` — this wording might make it clearer what's going on here:

```suggestion
clean-up is complete. :exc:`asyncio.CancelledError` directly subclasses
```

:exc:`BaseException` so most code will not need to be aware of it.

The asyncio components that enable structured concurrency, like
:class:`asyncio.TaskGroup` and :func:`asyncio.timeout`,
are implemented using cancellation internally and might misbehave if
a coroutine swallows :exc:`asyncio.CancelledError`. Similarly, user code
should not generally call :meth:`uncancel <asyncio.Task.uncancel>`.
However, in cases when suppressing :exc:`asyncio.CancelledError` is

> truly desired, it is necessary to also call :meth:`uncancel` to completely

Reviewer: This link doesn't work. I don't think we need it to work, since we link to `uncancel()` immediately above, but it will lead to a Sphinx warning being generated if the docs are built in "nitpicky mode" (#101100). Better to do this instead:

```suggestion
truly desired, it is necessary to also call ``uncancel()`` to completely
```

Author: Thanks, its been a long time since I worked with Python docs.

> :class:`asyncio.TaskGroup` and :func:`asyncio.timeout`,

Reviewer 2: Could I ask you to also rephrase "Most code can safely ignore :exc:`asyncio.CancelledError`" above? This vaguely seems like it would allow
```py
except CancelledError:
    pass  # Ignore CancelledError
```
which is the opposite of what it's trying to say.

Maybe say that most code doesn't need to be aware of it?

Author: How about the above?

are implemented using cancellation internally and might misbehave if
a coroutine swallows :exc:`asyncio.CancelledError`. Similarly, user code
should not generally call :meth:`uncancel <asyncio.Task.uncancel>`.

> However, in such cases when supprssing :exc:`asynci.CancelledError` is

Reviewer 2: Drop "such"
```suggestion
However, in cases when suppressing :exc:`asyncio.CancelledError` is
```

Reviewer 3: Mind the typo in `supprssing`
